oak-segment-azure/src/main/java/org/apache/jackrabbit/oak/segment/azure/AzureUtilities.java

```diff
@@ -116,6 +116,12 @@ public static CloudBlobDirectory cloudBlobDirectoryFrom(String connection, Strin
 
         return container.getDirectoryReference(dir);
     }
+    public static CloudBlobDirectory cloudBlobDirectoryFromSasUri(String connection, String containerName,
```

Comment:
Not sure this method is needed, since the only difference in this method compared to ```cloudBlobDirectoryFrom``` is that container creation is skipped. Otherwise in the method implementation I do not see anything specific to SAS URI.

Reply:
@mattvryan This class seem to be exported and is accessed from the `oak-upgrade` module. However, I don't think it's creating a security issue. This class just wraps the Azure SDK. If someone has valid credentials, then it could just as well use the Azure SDK directly to successfully connect.

Reply:
I agree with @smiroslav that this new method is misleading. What do we need is the ability to skip creating the container. I suggest to add a boolean parameter `skipCreateContainer` to the existing `cloudBlobDirectoryFrom()` methods. The value could then come from `MigrationOptions`, so that when running the upgrade tool, the flag can be set at the command line.